File: javascript/read_json_file.mjs
// Read JSON file, transform and print to stdout
import fs from 'fs/promises'

const jsonFile = process.argv[2]

const data = await fs.readFile(jsonFile, 'utf8')
const people = JSON.parse(data)

for (const person of people) {
  console.log(`Hello, ${person.age} year old ${person.first_name}`)
}

File: swift/read_json_file.swift
import Foundation

let jsonFile = "people.json"

let fileURL = URL(fileURLWithPath: jsonFile)
let jsonData = try Data(contentsOf: fileURL)
let people = try JSONSerialization.jsonObject(with: jsonData) as! [[String: Any]]

for (_, person) in people.enumerated() {
    let age = person["age"] as! Int
    let firstName = person["first_name"] as! String

    print("Hello, \(age) year old \(firstName)")
}
